Sync with HEAD
Fix "terminate called after throwing an instance of 'rrerror'"
fbx-faker depends on libX11 and libXext
--- a/ChangeLog.txt
+++ b/ChangeLog.txt
@@ -18,6 +18,11 @@ Transport with a remote X connection.
 Fixed a GL_INVALID_OPERATION error that would occur after a call to
 glXSwapBuffers() when a context with the OpenGL Core Profile was being used.
 -------------------------------------------------------------------------------
+[5]
+Fixed an issue whereby VirtualGL would abort with "terminate called after
+throwing an instance of 'rrerror'" whenever a 3D application running in
+VirtualGL exited and VirtualGL was compiled with GCC 4.6.
+-------------------------------------------------------------------------------
 
 
 ===============================================================================
diff --git a/include/rrmutex.h b/include/rrmutex.h
index 5a90475..ea49d90 100644
--- a/include/rrmutex.h
+++ b/include/rrmutex.h
@@ -178,10 +178,15 @@ class rrcs
 		class safelock
 		{
 			public:
-				safelock(rrcs &cs) : _cs(cs) {cs.lock();}
-				~safelock() {_cs.unlock();}
+				safelock(rrcs &cs, bool errorcheck=true) : _cs(cs),
+					_errorcheck(errorcheck)
+				{
+					cs.lock(errorcheck);
+				}
+				~safelock() {_cs.unlock(_errorcheck);}
 			private:
 				rrcs &_cs;
+				bool _errorcheck;
 		};
 
 	protected:
diff --git a/rr/fakerconfig.cpp b/rr/fakerconfig.cpp
index c3d8930..a121514 100644
--- a/rr/fakerconfig.cpp
+++ b/rr/fakerconfig.cpp
@@ -119,7 +119,7 @@ void fconfig_deleteinstance(void)
 {
 	if(fc!=NULL)
 	{
-		rrcs::safelock l(fcmutex);
+		rrcs::safelock l(fcmutex, false);
 		if(fc!=NULL)
 		{
 			#if FCONFIG_USESHM==1
diff --git a/util/CMakeLists.txt b/util/CMakeLists.txt
index 843eb6f..90791de 100644
--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -31,6 +31,7 @@ add_library(fbx STATIC fbx.c)
 if(VGL_BUILDSERVER)
 	add_library(fbx-faker STATIC fbx.c)
 	set_property(TARGET fbx-faker APPEND PROPERTY COMPILE_DEFINITIONS INFAKER)
+	target_link_libraries(fbx-faker ${X11_X11_LIB} ${X11_Xext_LIB})
 endif()
 
 add_executable(fbxtest fbxtest.cpp)
